target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4019-fritzbox-7530-toshiba.dts

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+
+#include "qcom-ipq4019-fritzbox-7530.dtsi"
+
+/ {
+	model = "AVM FRITZ!Box 7530 (Toshiba NAND)";
+	compatible = "avm,fritzbox-7530-toshiba", "avm,fritzbox-7530";
+
+};
+
+&nand {
+	pinctrl-0 = <&nand_pins>;
+	pinctrl-names = "default";
+	status = "okay";
+
+	nand@0 {
+		nand-ecc-strength = <8>;
+		nand-ecc-step-size = <512>;
```

Comment:
Is this the only difference between the two dts files?
Is it possible to use these settings on the older devices too?

Reply:
Yes. I took the VR9-based Fritz 3370 as template for the changes.
I don't think it will work on the older devices, but I don't have any to test. I think that the other NAND chips require 4 Bit ECC (in U-Boot, 4 Bit ECC is (was) hardcoded in the NAND controller driver and thus failed for this chip).
The (ugly) hack for U-Boot is here: https://github.com/chunkeey/FritzBox-4040-UBOOT/pull/8

Reply:
ok, thanks for the clarification